libs/sdk-ui-dashboard/src/presentation/topBar/menuButton/DefaultMenuButton.tsx

```diff
@@ -26,24 +28,151 @@ const bubbleAlignPoints: IAlignPoint[] = [{ align: "cl tr" }];
 export const DefaultMenuButton = (props: IMenuButtonProps): JSX.Element | null => {
```

Comment:
Isn't this a problem with cyclomatic complexity? This component is huge.

Possibly the blocks at the bottom such as `if (menuItem.type === "menu") ` can be extracted to a separate component/function as they only seem to take `menuItem` value, possibly one callback.

Reply:
splited